strawberryfields/apps/qchem/dynamics.py

```diff
@@ -250,3 +255,80 @@ def prob(samples: list, excited_state: list) -> float:
         raise ValueError("The excited state must not contain negative values")
 
     return samples.count(excited_state) / len(samples)
+
+
+def sample_coherent(
+    alpha: list, t: float, Ul: np.ndarray, w: np.ndarray, n_samples: int, loss: float = 0.0,
+) -> list:
+    r"""Generate samples for simulating vibrational quantum dynamics with an input Fock state.
+
+    **Example usage:**
+
+    >>> alpha = [0, 1.4]
+    >>> t = 10.0
+    >>> Ul = np.array([[0.707106781, -0.707106781],
+    >>>                [0.707106781, 0.707106781]])
+    >>> w = np.array([3914.92, 3787.59])
+    >>> n_samples = 5
+    >>> sample_coherent(alpha, t, Ul, w, n_samples)
+    [[0, 2], [0, 4], [0, 3], [0, 1], [0, 2]]
+
+    Args:
+        alpha (list): list of displacement parameters
+        t (float): time in femtoseconds
+        Ul (array): normal-to-local transformation matrix
+        w (array): normal mode frequencies :math:`\omega` in units of :math:`\mbox{cm}^{-1}`
+        n_samples (int): number of samples to be generated
+        loss (float): loss parameter denoting the fraction of lost photons
+
+    Returns:
+        list[list[int]]: a list of samples
+    """
+    if n_samples < 1:
```

Comment:
I'm not sure we need all of these checks. For example, the `LossChannel()` function should already throw an error if the loss parameter is invalid. Similarly, I think it's clear by context that time and frequencies must be positive

Reply:
It might still be good to keep checks on time and frequency values. I can remove the loss function check.

Reply:
Note that the `sample_fock` function still has those input checks.

Ideally we could have a quick `validation()` function that does all input validation, so we don't need to keep adding and writing tests for this in each function.

Reply:
That is a good suggestion, @trbromley. I will add one to `dynamics.py`